Src/FluentAssertions/Collections/GenericDictionaryAssertions.cs

```diff
@@ -242,6 +243,87 @@ public GenericDictionaryAssertions(IDictionary<TKey, TValue> dictionary) : base(
             return new AndConstraint<GenericDictionaryAssertions<TKey, TValue>>(this);
         }
 
+        /// <summary>
+        /// Assert that the current dictionary has the same number of elements as <paramref name="otherCollection" />.
+        /// </summary>
+        /// <param name="otherCollection">The other collection with the same expected number of elements</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public AndConstraint<GenericDictionaryAssertions<TKey, TValue>> HaveSameCount(IEnumerable otherCollection, string because = "",
```

Comment:
Shouldn't this method take a dictionary?

Reply:
That would prevent you from writing the following test

```c#
// Arrange
var subject = new List<int> { 1, 2, 3};

// Act
Dictionary<int, int> actual = ConvertToDictionary(subject);

// Assert
actual.Should().HaveSameCount(subject);
```

Reply:
Agreed, this is the main scenario for me, converting to/from dictionaries and then validating the results.